tests/printing/transformed/lazy-vals-new.check

```diff
@@ -0,0 +1,106 @@
+    @volatile lazy <static> var x$lzy1: Object = null
+    lazy def x(): Int = 
+      {
+        var result: Object = A#x$lzy1
+        if result.isInstanceOf[Int] then return scala.Int.unbox(result) else 
+          if result.==(scala.runtime.LazyVals.NullValue) then 
+            return scala.Int.unbox(null)
+           else return scala.Int.unbox(A.x$lzyINIT1())
+      }
+    private def x$lzyINIT1(): Object = 
+      while <empty> do 
+        {
+          val current: Object = A#x$lzy1
+          if current.==(null) then 
+            if 
+              scala.runtime.LazyVals.objCAS(classOf[A], A.OFFSET$_m_0, null, 
+                scala.runtime.LazyVals.Evaluating
+              )
+             then 
+              {
+                var result: Object = null
+                var resultNullable: Object = null
+                try 
+                  {
+                    resultNullable = scala.Int.box(2)
+                    if resultNullable.==(null) then 
+                      result = scala.runtime.LazyVals.NullValue
+                     else result = resultNullable
+                    return resultNullable
+                  }
+                 finally 
+                  {
+                    if 
+                      scala.runtime.LazyVals.objCAS(classOf[A], A.OFFSET$_m_0, 
+                        scala.runtime.LazyVals.Evaluating
+                      , result).unary_!()
+                     then 
+                      {
+                        val lock: scala.runtime.LazyVals.LazyVals$Waiting = 
+                          A#x$lzy1.asInstanceOf[
+                            scala.runtime.LazyVals.LazyVals$Waiting
+                          ]
+                        scala.runtime.LazyVals.objCAS(classOf[A], A.OFFSET$_m_0
+                          , 
+                        lock, result)
+                        lock.countDown()
+                      }
+                     else ()
+                    ()
+                  }
+              }
+             else ()
+           else 
+            if 
+              current.isInstanceOf[
+                scala.runtime.LazyVals.LazyVals$LazyValControlState
+              ]
+             then 
+              if current.==(scala.runtime.LazyVals.Evaluating) then 
+                {
+                  scala.runtime.LazyVals.objCAS(classOf[A], A.OFFSET$_m_0, 
+                    scala.runtime.LazyVals.Evaluating
+                  , new scala.runtime.LazyVals.LazyVals$Waiting())
+                  ()
+                }
+               else 
+                if current.isInstanceOf[scala.runtime.LazyVals.LazyVals$Waiting]
+                   then
+                 
+                  current.asInstanceOf[scala.runtime.LazyVals.LazyVals$Waiting].
+                    await
+                  ()
+                 else 
+                  if current.==(scala.runtime.LazyVals.NullValue) then 
+                    return null
+                   else ()
```

Comment:
It can't be anything else than `Evaluating`, `Waiting` or `NullValue`, can it? If not, consider removing the `if current == NullValue` and directly do `return null` instead.